noir-projects/noir-protocol-circuits/crates/blob/src/blob_batching.nr

```diff
@@ -94,14 +96,38 @@ pub fn evaluate_blobs_and_batch(
     end_accumulator
 }
 
+// Validates this block's injected blob fields against the validated sponge propagated from the previous circuits:
+// - Checks that we haven't accumulated too many fields over the rollup
+// - Checks that we have absorbed the exact number of fields we claim to have accumulated
+// - Checks that the injected fields match the tx effects absorbed over the rollup
+// - Checks that any fields above expected_fields are empty (inside poseidon2_hash_subarray())
+fn check_block_blob_sponge(
+    blobs_as_fields: [Field; FIELDS_PER_BLOB * BLOBS_PER_BLOCK],
+    mut sponge_blob: SpongeBlob,
+) -> Field {
```

Comment:
This (and `compute_blob_challenge`) has been moved from `blob.nr` to keep all non-evaulation checks in one place.